### PR TITLE
Add AYON APIs & Resources section

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
This PR adds AYON APIs & Resources to Docs.
It adds two things: 
- Add `Dev resources` drop down menu in the navigation bar.
-  Add `APIs & Resources` section in dev docs. where we can explain what is a particular API is used for and how to use it.

image

### TODOs
Write a brief description about each item in this list, as well as add some examples: 
- [x] REST API Docs
- [x] GraphQL API Docs
- [x] Python API Docs
- [x] C++ API Docs
- [x] USD Resolver API Docs
- [x] Frontend React Components

### CODE REVIEW

**Comment by mkolar** [comment]
I really like it, however `APIs and components` might be a bit too specific. How about calling it `Dev resources`?

**Comment by MustafaJafar** [comment]
> I really like it, however `APIs and components` might be a bit too specific. How about calling it `Dev resources`?

I like `Dev resources` more. 
I changed it in the navbar. 
image

File: website/sidebars.js [hunk 1]
```diff
@@ -196,6 +196,25 @@ module.exports = {
                 "dev_launcher_build_linux",
             ],
         },
+        {
+            type: "category",
+            label: "AYON APIs & Components",
```
Comment (mkolar) [inline website/sidebars.js:201]:
```suggestion
            label: "APIs & Resources",
```

File: website/sidebars.js [hunk 2]
```diff
@@ -196,6 +196,25 @@ module.exports = {
...
+            link: {
+                type: 'generated-index',
+                title: 'AYON APIs and Components Catalogue',
```
Comment (mkolar) [inline website/sidebars.js:204]:
```suggestion
                title: 'AYON APIs an Developer resources',
```

File: website/sidebars.js [hunk 3]
```diff
@@ -196,6 +196,25 @@ module.exports = {
...
+                description:
+                  "Welcome to AYON APIs and Components Catalogue!",
```
Comment (mkolar) [inline website/sidebars.js:206]:
```suggestion
                  "Welcome to AYON APIs and development resources catalogue!",
```

**Review by mkolar — CHANGES_REQUESTED** [review]

**Review by mkolar — APPROVED** [review]

**Comment by MustafaJafar** [comment]
Then, the next step begins. Let's finish these TODOs 🛠️

File: website/docs/dev_api_cpp.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
+---
+id: dev_api_cpp
+title: AYON C++ API
+sidebar_label: C++ API
+description: Information about AYON C++ API
+toc_max_heading_level: 5
+---
+
+What is the C++ Language standard? it's ISO C++17 Standard
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_cpp.md:9]: Having this sentence on first line seems weird (out of context).

File: website/docs/dev_api_python.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+id: dev_api_python
+title: AYON Python API
+sidebar_label: Python API
+description: Information about AYON Python API
+toc_max_heading_level: 5
+---
+
+## About AYON Python API
+
+A Python client for connecting to AYON server. The client is using REST and GraphQl to communicate with server with `requests` module.
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_python.md:11]: Not sure if matters, but graphql is in fact working via REST api too. It is REST endpoint.

File: website/docs/dev_api_python.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+The module support singleton connection which is using `AYON_SERVER_URL` and `AYON_API_KEY` environment variables as source for connection.
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_python.md:13]:
```suggestion
The module supports singleton connection which is using `AYON_SERVER_URL` and `AYON_API_KEY` environment variables as source for connection. But you can create multiple connection to different servers using `ServerAPI` class.
```

File: website/docs/dev_api_cpp.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+## About AYON C++ API
+
+The AYON CPP API allows communication with the AYON server.
+Its focus is on communication with the server Resolve endpoint.
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_cpp.md:14]:
```suggestion
Its focus is on communication with the server resolve endpoint.
```

File: website/docs/dev_api_cpp.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+and it's meant to be used mainly by [AYON USD Resolver](dev_api_usd_resolver.md).
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_cpp.md:15]:
```suggestion
Implementation is mainly focused on [AYON USD Resolver](dev_api_usd_resolver.md).
```

File: website/docs/dev_api_cpp.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+## Get Started
+
+:::tip Development Environment
+We assume you already had your c++ dev environment setup as well as cmake.
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_cpp.md:20]:
```suggestion
We assume you already have setup your c++ dev environment as well as cmake.
```

File: website/docs/dev_api_cpp.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+:::
+
+### Fetch and Build The library
+AYON CPP API is a static lib so you'd need to build it first.
+
+You can use the provided python script to do the hard lifting for you.
```
Comment (iLLiCiTiT) [inline website/docs/dev_api_cpp.md:26]:
```suggestion
AYON CPP API is a static lib.

You need to build it first. We've prepared python script to do the hard lifting for you.
```

Comment (MustafaJafar) [reply website/docs/dev_api_cpp.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/b534281136c835274d3343536b40171be9f9bca0

File: website/docs/dev_api_cpp.md (hunk 9 above)
Comment (MustafaJafar) [reply website/docs/dev_api_cpp.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/b534281136c835274d3343536b40171be9f9bca0

File: website/docs/dev_api_cpp.md (hunk 8 above)
Comment (MustafaJafar) [reply website/docs/dev_api_cpp.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/b534281136c835274d3343536b40171be9f9bca0

File: website/docs/dev_api_cpp.md (hunk 4 above)
Comment (MustafaJafar) [reply website/docs/dev_api_cpp.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/b534281136c835274d3343536b40171be9f9bca0

File: website/docs/dev_api_python.md (hunk 5 above)
Comment (MustafaJafar) [reply website/docs/dev_api_python.md:11]: As far as I know, the implementation details of graphQL, the way it works and its resolvers is different from the fact that it provides a different communication channel with AYON Server. 
and `ayon-python-api` is using both REST API and GraphQl calls.

what do you think?

File: website/docs/dev_api_python.md (hunk 6 above)
Comment (MustafaJafar) [reply website/docs/dev_api_python.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/229ec8af99f71db9a12efe7d2da1462234fc3023

File: website/docs/dev_api_python.md (hunk 5 above)
Comment (iLLiCiTiT) [reply website/docs/dev_api_python.md:11]: A Python client for connecting to AYON server using `requests` module. Wraps REST API endpoints including GraphQl to communicate with server.

Comment (MustafaJafar) [reply website/docs/dev_api_python.md]: https://github.com/ynput/ayon-documentation/pull/239/commits/90f6267567c3e36d417f0d337df9bc43f345b934

**Comment by Lypsolon** [comment]
for repos with submodules, you should clone via `git clone --recurse-submodules <URL>`

**Comment by mkolar** [comment]
@MustafaJafar how is this looking?

**Review by mkolar — APPROVED** [review]

**Comment by MustafaJafar** [comment]
> @MustafaJafar how is this looking?

It looks good on my side and all examples were tested.

**Pull request merged**